test/plugins/validator_test.rb

```diff
@@ -134,6 +134,44 @@ def test_optional_parameter_without_a_value_fails_validation
   end
 end
 
+class OptionalUrlValidatorTest < Test::Unit::TestCase
+  class OptionalUrlValidatorTestPlugin < ::Proxy::Plugin
+    default_settings url: 'http://example.com'
+  end
+
+  def validator
+    ::Proxy::PluginValidators::Url.new(OptionalUrlValidatorTestPlugin, 'url', nil, nil)
+  end
+
+  def test_required_parameter_with_a_value_passes_validation
+    assert validator.validate!(url: 'http://example.com')
+  end
+
+  def test_empty_string_treated_as_missing_value
+    error = assert_raises ::Proxy::Error::ConfigurationError do
+      validator.validate!(url: '')
+    end
+
+    assert_match(/expected to contain a url/, error.message)
+  end
+
+  def test_without_a_value_fails_validation
+    error = assert_raises ::Proxy::Error::ConfigurationError do
+      validator.validate!(url: nil)
+    end
+
+    assert_match(/expected to contain a url/, error.message)
+  end
```

Comment:
I'd have expected this to actually verify nil was accepted somewhere. If this fails validation, it doesn't sound like it's working.